src/config.c

```diff
@@ -588,7 +589,9 @@ void loadServerConfigFromString(char *config) {
                 queueSentinelConfig(argv+1,argc-1,linenum,lines[i]);
             }
         } else {
-            err = "Bad directive or wrong number of arguments"; goto loaderr;
+            err = "Bad directive or wrong number of arguments";
+            sdsfreesplitres(argv,argc);
+            goto loaderr;
```

Comment:
Aren't every other `goto loaderr` a problem? I would rather us add a check in loaderr saying `if (argv) sdsfreesplitres(argv,argc);` Then add all of these on every line.

Reply:
Hi @madolson ,
I agree with you, I have already started checking for other cases. I will add as you mentioned and test.

Reply:
can't we put the cleanup in the `loaderr` label, and nullify something in the (i assume fewer) other places that we call `sdsfreesplitres`). this will result in fewer changes, and safer code.

Reply:
Hi @oranagra ,

Reply:
Hi @oranagra ,

argv and argc are declared in for loop so if we try to free in loaderr, pointer and variable is going out of scope. In order to do we have to bring argv and argc to function scope. I feel its safe to bring them, I will change scope and test all the cases, if everything works fine will update the diff.  Thanks!

Reply:
We can move the declaration up outside of the loop.